### PR TITLE
Removed global state

### PR DESCRIPTION
**Pull request by McCallisterRomer**
* Replaced the majority of global pointers with abstract classes accessible from the NcEngine base class.
  * Internal classes had to be changed slightly to implement the new interfaces. Some names were also changed to fit a consistent scheme: XXXSystem refers to an API facing type and XXXSystemImpl refers to the internal implementer.
  * Scene::Load was changed to take NcEngine* instead of registry_type*.
  * There are at least two remaining cases to fix: ActiveRegistry() and g_instance for WindowImpl.
* Most nested namespaces were removed from /include. These make sense in /source where there are substantial, self-contained modules like nc::physics, but those lines aren't as clear in /include.
* Moved view/projection/frustum calculations into the Camera component.

### CODE REVIEW

**Comment by ClickerMonkey** [comment]
Thats a great improvement!

File: nc/source/engine/Engine.cpp [hunk 1]
```diff
@@ -76,11 +43,47 @@ namespace nc
...
+    }
+
+    void Engine::Shutdown() noexcept
+    {
+        V_LOG("Shutdown NcEngine");
+        ClearState();
+        debug::internal::CloseLog();
+    }
+
+    auto Engine::Audio()       noexcept -> AudioSystem*     { return & m_audioSystem;     }
```
Comment (jaremieromer) [inline nc/source/engine/Engine.cpp]: extra space

File: project/source/shared/EdgePanCamera.cpp [hunk 2]
```diff
@@ -43,6 +40,7 @@ namespace nc::sample
             translation = Vector3::Zero();
         m_lastFrameTranslation = translation;
         translation = translation * dt;
-        camera::GetMainCameraTransform()->Translate(translation);
+        //GetMainCameraTransform()->Translate(translation);
```
Comment (jaremieromer) [inline project/source/shared/EdgePanCamera.cpp]: Remove

File: project/source/shared/SceneNavigationCamera.h [hunk 3]
```diff
@@ -35,7 +35,8 @@ namespace nc::sample
 
     inline void SceneNavigationCamera::FrameUpdate(float dt)
     {
-        auto* transform = camera::GetMainCameraTransform();
+        auto* transform = ActiveRegistry()->Get<Transform>(GetParentEntity());
+        //auto* transform = GetMainCameraTransform();
```
Comment (jaremieromer) [inline project/source/shared/SceneNavigationCamera.h]: Remove

**Review by jaremieromer — APPROVED** [review]

**Pull request merged**